iis-workflows/iis-workflows-transformers/src/main/java/eu/dnetlib/iis/workflows/transformers/spark/citationmatching/direct/DocumentToDirectCitationMetadataConverter.java

```diff
@@ -0,0 +1,63 @@
+package eu.dnetlib.iis.workflows.transformers.spark.citationmatching.direct;
+
+import java.io.Serializable;
+import java.util.List;
+
+import com.google.common.collect.Lists;
+
+import eu.dnetlib.iis.citationmatching.direct.schemas.DocumentMetadata;
+import eu.dnetlib.iis.citationmatching.direct.schemas.ReferenceMetadata;
+import eu.dnetlib.iis.transformers.metadatamerger.schemas.ExtractedDocumentMetadataMergedWithOriginal;
+
+/**
+ * Converter of {@link ExtractedDocumentMetadataMergedWithOriginal} object
+ * to {@link DocumentMetadata} object
+ * 
+ * @author madryk
+ *
+ */
+public class DocumentToDirectCitationMetadataConverter implements Serializable {
+
+	private static final long serialVersionUID = 1L;
+
+
+	//------------------------ LOGIC --------------------------
+
+	/**
+	 * Converts {@link ExtractedDocumentMetadataMergedWithOriginal} to {@link DocumentMetadata}
+	 */
+	public DocumentMetadata convert(ExtractedDocumentMetadataMergedWithOriginal docMetadata) {
+		List<ReferenceMetadata> citationReferencesMetadata = Lists.newArrayList();
+
+		if (docMetadata.getReferences() != null) {
+			for (eu.dnetlib.iis.metadataextraction.schemas.ReferenceMetadata docReferenceMetadata : docMetadata.getReferences()) {
+
+				ReferenceMetadata citationReferenceMetadata = convertReference(docReferenceMetadata);
+
+				if (citationReferenceMetadata != null) {
+					citationReferencesMetadata.add(citationReferenceMetadata);
+				}
+			}
+		}
```

Comment:
The 'if' here... it would be nice to move it a private method named e.g. convertReferences(docMetadata.getReferences()) returning List<ReferenceMetadata> citationReferencesMetadata. What do you think?